src/origin_sdk/service/Project.py

```diff
@@ -20,10 +20,90 @@ def get(self, key: str):
         """
         return self.project.get(key)
 
+    def refresh(self):
+        """Refreshes the project data."""
+        self.project = self.session.get_project(self.project_id)
+        return self
+
     def pin(self):
+        """Pins the project, if it isn't already"""
         if not self.project.get("isProjectPinned"):
             self.session.pin_project(self.project_id)
+        return self
 
     def unpin(self):
+        """Unpins the project, if it isn't already"""
         if self.project.get("isProjectPinned"):
             self.session.unpin_project(self.project_id)
+        return self
+
+    def get_or_create_scenario_by_name(
+        self, scenario_name: str, base_scenario_id: Optional[str] = None
```

Comment:
I think this should be two methods. I understand it's like an 'upsert' equivalent but I think the optional base_scenario_id parameter makes it basically 2 functions